modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/segstore/IndexFileManager.java

```diff
@@ -120,20 +129,36 @@ class IndexFileManager {
         Files.createDirectories(indexFilesDir);
     }
 
+    void start() throws IOException {
+        try (Stream<Path> indexFiles = Files.list(indexFilesDir)) {
+            Iterator<Path> it = indexFiles.sorted().iterator();
+
+            while (it.hasNext()) {
+                recoverIndexFileMetas(it.next());
+            }
+        }
+    }
+
     Path indexFilesDir() {
         return indexFilesDir;
     }
 
     /**
      * Saves the given index memtable to a file.
+     *
+     * <p>Must only be called by the checkpoint thread.
      */
     Path saveIndexMemtable(ReadModeIndexMemTable indexMemTable) throws IOException {
-        String fileName = indexFileName(curFileOrdinal, 0);
+        return saveIndexMemtable(indexMemTable, ++curFileOrdinal);
+    }
+
+    Path saveIndexMemtable(ReadModeIndexMemTable indexMemTable, int fileOrdinal) throws IOException {
+        String fileName = indexFileName(fileOrdinal, 0);
```

Comment:
I know I had asked this in an earlier PR, but just to refresh my memory, why is the generation always set to 0 here?